### PR TITLE
Implement MultiIndex.levshape

### PR DESCRIPTION
**Pull request by RainFung**
Get MultiIndex unique value count in every level.

### CODE REVIEW

File: databricks/koalas/indexes.py [hunk 1]
```diff
@@ -807,6 +807,27 @@ def name(self) -> str:
...
+        --------
+        >>> midx = ks.MultiIndex.from_tuples([('a', 'x'), ('b', 'y'), ('c', 'z')])
+        >>> midx  # doctest: +SKIP
+        MultiIndex([('a', 'x'),
+                    ('b', 'y'),
+                    ('c', 'z')],
+                   )
+
+        >>> midx.levshape
+        (3,3)
```
Comment (itholic) [inline databricks/koalas/indexes.py]: i think we skip this also.

Seems that it have different print out behavior depends on python version.

`(3,3)` for 3.6, but `(3, 3)` for 3.5 & 3.7

Comment (RainFung) [reply databricks/koalas/indexes.py]: Ok, it's a common issue in tuple.

**Comment by RainFung** [comment]
Tuple issue caused by python version need help.

File: databricks/koalas/indexes.py [hunk 2]
```diff
@@ -832,6 +832,27 @@ def name(self) -> str:
...
+        Examples
+        --------
+        >>> midx = ks.MultiIndex.from_tuples([('a', 'x'), ('b', 'y'), ('c', 'z')])
+        >>> midx  # doctest: +SKIP
+        MultiIndex([('a', 'x'),
+                    ('b', 'y'),
+                    ('c', 'z')],
+                   )
+
+        >>> midx.levshape
```
Comment (HyukjinKwon) [inline databricks/koalas/indexes.py]: Can you skip this test then? Seems the results varies due to Python version as pointed out. You can do `  # doctest: +SKIP`

File: databricks/koalas/indexes.py [hunk 3]
```diff
@@ -831,6 +831,27 @@ def name(self) -> str:
     def name(self, name: str) -> None:
         raise PandasNotImplementedError(class_name='pd.MultiIndex', property_name='name')
 
+    @property
+    def levshape(self) -> tuple:
```
Comment (HyukjinKwon) [inline databricks/koalas/indexes.py]: Seems you should remove `levshape` in `_MissingPandasLikeMultiIndex`

**Comment by RainFung** [comment]
@itholic  Can you help me solve the unknown failing problem? Thanks.

File: databricks/koalas/missing/indexes.py [hunk 4]
```diff
@@ -119,7 +119,7 @@ class _MissingPandasLikeMultiIndex(object):
 
     # Properties
     is_all_dates = unsupported_property('is_all_dates')
-    levshape = unsupported_property('levshape')
+    shape = unsupported_property('shape')
```
Comment (itholic) [inline databricks/koalas/missing/indexes.py]: why do you add this?? i thnik it is already implemented ?

```python
>>> index = Index([1, 2, 3])
>>> index.shape
(3,)
```

not pretty sure, but i guess maybe that is the reason of build failure  :)

Comment (itholic) [reply databricks/koalas/missing/indexes.py]: in the above question, "method for checking the python version" means like that?

```python
>>> import sys
>>> sys.version
'3.7.2 (v3.7.2:9a3ffc0492, Dec 24 2018, 02:44:43) \n[Clang 6.0 (clang-600.0.57)]'
```

File: databricks/koalas/indexes.py (hunk 1 above)
Comment (ueshin) [reply databricks/koalas/indexes.py]: What's going on here?
I don't think the tuple string is not changed throughout 3.5 - 3.7.

```py
Python 3.5.6 |Anaconda, Inc.| (default, Aug 26 2018, 16:30:03)
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> (3,3)
(3, 3)
```

```py
Python 3.6.9 |Anaconda, Inc.| (default, Jul 30 2019, 13:42:17)
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> (3,3)
(3, 3)
```

```py
Python 3.7.5 (default, Oct 25 2019, 10:52:18)
[Clang 4.0.1 (tags/RELEASE_401/final)] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> (3,3)
(3, 3)
```

File: databricks/koalas/tests/test_indexes.py [hunk 5]
```diff
@@ -150,6 +151,14 @@ def test_multi_index_names(self):
         with self.assertRaises(PandasNotImplementedError):
             kidx.name = 'renamed'
 
+    def test_multi_index_levshape(self):
+        pidx = pd.MultiIndex.from_tuples([('a', 'x', 1), ('b', 'y', 2)])
+        kidx = ks.MultiIndex.from_tuples([('a', 'x', 1), ('b', 'y', 2)])
+        if sys.version[0:3] == '3.6':
+            self.assertEqual(pidx.levshape, kidx.levshape)
+        else:
+            self.assertEqual((3, 3), kidx.levshape)
```
Comment (ueshin) [inline databricks/koalas/tests/test_indexes.py]: Why do we need to special handle 3.6?

```py
Python 3.6.8 |Anaconda, Inc.| (default, Dec 29 2018, 19:04:46)
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pandas as pd
>>> pd.MultiIndex.from_tuples([('x', 'a'), ('x', 'b')]).levshape
(1, 2)
```

```py
Python 3.7.5 (default, Oct 25 2019, 10:52:18)
[Clang 4.0.1 (tags/RELEASE_401/final)] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pandas as pd
>>> pd.MultiIndex.from_tuples([('x', 'a'), ('x', 'b')]).levshape
(1, 2)
```

seems same?

File: databricks/koalas/indexes.py (hunk 1 above)
Comment (ueshin) [reply databricks/koalas/indexes.py]: It should be `(3, 3)` and we should not skip it.

File: databricks/koalas/indexes.py [hunk 6]
```diff
@@ -831,6 +831,27 @@ def name(self) -> str:
...
+        MultiIndex([('a', 'x'),
+                    ('b', 'y'),
+                    ('c', 'z')],
+                   )
+
+        >>> midx.levshape # doctest: +SKIP
+        (3,3)
+        """
+        result = self._internal._sdf\
+            .agg(*(F.countDistinct(F.col(c)) for c in self._internal.index_columns)).collect()
```
Comment (ueshin) [inline databricks/koalas/indexes.py]: `self._internal.index_scols` instead of `self._internal.index_columns`?

File: databricks/koalas/indexes.py [hunk 7]
```diff
@@ -831,6 +831,27 @@ def name(self) -> str:
...
+        return tuple(v for v in result[0])
```
Comment (ueshin) [inline databricks/koalas/indexes.py]: `tuple(result[0])` should work?

File: databricks/koalas/indexes.py (hunk 1 above)
Comment (ueshin) [reply databricks/koalas/indexes.py]: ah, I see, the travis tests for the py3.6 environment are wrong.

Comment (ueshin) [reply databricks/koalas/indexes.py]: Submitted a PR #1107.

Comment (itholic) [reply databricks/koalas/indexes.py]: Ah, that was a problem. Thanks for fixing @ueshin :+1:

File: databricks/koalas/indexes.py [hunk 8]
```diff
@@ -831,6 +831,27 @@ def name(self) -> str:
...
+        Examples
+        --------
+        >>> midx = ks.MultiIndex.from_tuples([('a', 'x'), ('b', 'y'), ('c', 'z')])
+        >>> midx  # doctest: +SKIP
+        MultiIndex([('a', 'x'),
+                    ('b', 'y'),
+                    ('c', 'z')],
+                   )
+
+        >>> midx.levshape # doctest: +SKIP
```
Comment (ueshin) [inline databricks/koalas/indexes.py]: Could you remove `# doctest: +SKIP`? It should work.

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**